main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -471,7 +471,7 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
 
   object Expressions {
```

Comment:
This looks could, but the way you implemented the changes requires a million changes all over. Here is a simpler solution:

- Do not change Expression to Block globally.
- Let Expression go to Block instead of to Statement.
- Leave Expression.Statement as-is, and only use it where you want to allow it, e.g. in let, match, and def.

Reply:
Does that make sense?

Reply:
Yeah, Thats a better way to do it.